### PR TITLE
Nameserver01: Add unit tests for existing messages

### PR DESCRIPTION
**Pull request by mattias-p**
Partial implementation of dotse/zonemaster#625.

Adds unit tests for existing checks/messages.

### CODE REVIEW

File: t/Test-nameserver01-B.t [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+if ( not $ENV{ZONEMASTER_RECORD} ) {
+    die q{Stored data file missing} if not -r $datafile;
+    Zonemaster::Engine::Nameserver->restore( $datafile );
+    Zonemaster::Engine->config->no_network( 1 );
+}
+
+Zonemaster::Engine->add_fake_delegation(
+    'b.nameserver01.exempelvis.se.' => {
+        'ns1.b.nameserver01.exempelvis.se' => [ '46.21.97.97', '2a02:750:12:77::97' ],
+        'ns2.b.nameserver01.exempelvis.se' => [ '46.21.97.97', '2a02:750:12:77::97' ],
```
Comment (matsduf) [inline t/Test-nameserver01-B.t]: Is it correct that both servers have the same IP here?

Comment (mattias-p) [reply t/Test-nameserver01-B.t]: `dig @ns3.binero.se b.nameserver01.exempelvis.se SOA` gives me a delegation to the parent zone with name servers bill.ip.se and bull.ip.se. `dig @bull.ip.se b.nameserver01.exempelvis.se SOA` gives me the above delegation with duplicated addresses. Maybe I shouldn't have implemented this before you validated the test zone?

File: t/Test-nameserver01-B.t [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+if ( not $ENV{ZONEMASTER_RECORD} ) {
+    die q{Stored data file missing} if not -r $datafile;
+    Zonemaster::Engine::Nameserver->restore( $datafile );
+    Zonemaster::Engine->config->no_network( 1 );
+}
+
+Zonemaster::Engine->add_fake_delegation(
+    'b.nameserver01.exempelvis.se' => {
+        'ns1.b.nameserver01.exempelvis.se' => [ '46.21.97.97', '2a02:750:12:77::97' ],
+        'ns2.b.nameserver01.exempelvis.se' => [ '46.21.97.97', '2a02:750:12:77::97' ],
```
Comment (matsduf) [inline t/Test-nameserver01-B.t]: The data seems to be wrong. Correct data should be

```
;; AUTHORITY SECTION:
b.nameserver01.exempelvis.se. 3600 IN	NS	ns1.b.nameserver01.exempelvis.se.
b.nameserver01.exempelvis.se. 3600 IN	NS	ns2.b.nameserver01.exempelvis.se.

;; ADDITIONAL SECTION:
ns1.b.nameserver01.exempelvis.se. 3600 IN A	46.21.97.97
ns1.b.nameserver01.exempelvis.se. 3600 IN AAAA	2a02:750:12:77::97
ns2.b.nameserver01.exempelvis.se. 3600 IN A	37.123.169.91
ns2.b.nameserver01.exempelvis.se. 3600 IN AAAA	2001:9b0:1:1c13::53
```

**Comment by matsduf** [comment]
Do a review too early? "B" data has been fixed, but "D" data is still missing.

**Comment by mattias-p** [comment]
Now it's complete. Please review.

**Pull request merged**